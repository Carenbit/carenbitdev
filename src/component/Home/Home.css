#homePage {
  display: grid;
  grid-template-rows: 0.5fr 1fr;
  grid-template-columns: 0.8fr 1fr;
  gap: 0px;
  padding-left: 10%;
  padding-top: 10%;
  background: url("../../assets/heroEllipse.png");
  background-position: top right;
  background-repeat: no-repeat;
  background-size: contain;
  max-height: 90vh;
}

.plmImage {
  grid-row-start: 1;
  grid-column-start: 2;
  grid-row-end: 3;
  grid-column-end: 3;
}

.mainHeading {
  font-family: Inter;
  font-size: 40px;
  font-weight: 800;
  line-height: 48px;
  letter-spacing: -0.005em;
  padding-top: 15%;
  height: fit-content;
  padding-bottom: 20px;
}

p {
  font-size: 22px;
  font-weight: 400;
  line-height: 30px;
  letter-spacing: -0.02em;
}

@media screen and (max-width: 464px) {
  #homePage {
    grid-template-rows: 0.2fr 0.5fr;
    grid-template-columns: 1fr;
    text-align: center;
    padding-top: 0;
    padding-right: 10%;
    background-position-y: 60px;
    background: size 80%;
  }

  .plmImage {
    grid-row-start: 2;
    grid-column-start: 1;
    grid-row-end: 3;
    grid-column-end: 2;
  }

  .mainHeading {
    font-size: 30px;
    line-height: 40.5px;
    padding-bottom: 20px;
    padding-left: 10%;
    text-align: center;
    max-width: 300px;
  }

  p {
    font-size: 16px;
    max-width: 300px;
  }
}

@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) {
  #homePage {
    grid-template-rows: 0.3fr 0.5fr;
    grid-template-columns: 0.8fr 1fr;
    padding-top: 15%;
  }

  .mainHeading {
    font-size: 30px;
    line-height: 36px;
    letter-spacing: -0.005em;
    padding-top: 15%;
    height: fit-content;
    padding-bottom: 20px;
  }
  p {
    font-size: 16px;
  }

  .plmImage {
    padding-left: 10%;
  }
}

@media only screen and (min-device-width: 464px) and (max-device-width: 768px) {
  #homePage {
    grid-template-rows: 1fr 1fr;
    grid-template-columns: 0.8fr 1fr;
    padding-top: 10%;
  }

  .plmImage {
    padding-top: 25%;
  }

  .mainHeading {
    font-size: 26px;
    line-height: 36px;
    padding-top: 15%;
    height: fit-content;
  }
  p {
    font-size: 16px;
  }
}

@media only screen and (min-device-width: 1024px) {
  .plmImage {
    padding-left: 20%;
  }
}
