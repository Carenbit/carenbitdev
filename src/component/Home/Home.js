import React from "react";
import "./Home.css";
import { MdEmail } from "react-icons/md";
import frameHome from "../../assets/frameHome.png";

const Home = () => {
  return (
    <div id="homePage">
      <h1 class="mainHeading">Simplify Your Product Journey with Carenbit</h1>
      <div>
        <p>
          Explore Carenbit's PLM Solutions for Enhanced Product Lifecycle
          Management.
        </p>
        <a href="mailto:companyxyz@gmail.com" target="_blank" rel="noreferrer">
          <button>
            <MdEmail className="mailIcon" />
            Quick Enquiry
          </button>
        </a>
      </div>
      <img src={frameHome} alt="..." class="plmImage img-fluid" />
    </div>
  );
};

export default Home;
