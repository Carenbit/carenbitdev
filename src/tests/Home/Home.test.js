import "@testing-library/jest-dom";
import Home from "../../component/Home/Home.js";
import { render, screen } from "@testing-library/react";

test("renders Home component", () => {
  render(<Home />);
  const linkElement = screen.getAllByText(/Carenbit/i);
  expect(linkElement).toHaveLength(2);
});

test("should have a quick enquiry button", () => {
  render(<Home />);
  const button = screen.getByText(/Quick Enquiry/i);
  expect(button).toBeInTheDocument();
  expect(
    screen.getByRole("button", { name: /Quick Enquiry/i })
  ).toBeInTheDocument();
});

test("should match screenshot of Home component", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
