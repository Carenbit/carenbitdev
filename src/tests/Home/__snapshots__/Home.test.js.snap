// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot of Home component 1`] = `
<div>
  <div
    id="homePage"
  >
    <h1
      class="mainHeading"
    >
      Simplify Your Product Journey with Carenbit
    </h1>
    <div>
      <p>
        Explore Carenbit's PLM Solutions for Enhanced Product Lifecycle Management.
      </p>
      <a
        href="mailto:carenbit.tech@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        <button>
          <svg
            class="mailIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          Quick Enquiry
        </button>
      </a>
    </div>
    <img
      alt="..."
      class="plmImage img-fluid"
      src="frameHome.png"
    />
  </div>
</div>
`;
